### PR TITLE
Add additional UMA metrics to AppendBuffer

### PR DESCRIPTION
**Pull request by at-ninja**
In addition to latency data, the size of the data being appended, along with the rate of speed that the data was processed will be recorded.

b/340663759

### CODE REVIEW

File: cobalt/dom/source_buffer_metrics.cc [hunk 1]
```diff
@@ -121,21 +121,33 @@ void SourceBufferMetrics::EndTracking(SourceBufferMetricsAction action,
...
           base::TimeDelta::FromMilliseconds(25), 50);
       break;
-    case SourceBufferMetricsAction::APPEND_BUFFER:
+    case SourceBufferMetricsAction::APPEND_BUFFER: {
       UMA_HISTOGRAM_CUSTOM_MICROSECONDS_TIMES(
           "Cobalt.Media.SourceBuffer.AppendBuffer.Timing", action_duration,
           base::TimeDelta::FromMicroseconds(1),
           base::TimeDelta::FromMilliseconds(25), 50);
+
+      UMA_HISTOGRAM_COUNTS_10M(
```
Comment (joeltine) [inline cobalt/dom/source_buffer_metrics.cc:140]: Can you look at the UMA_HISTOGRAM_MEMORY* set of macros and verify you don't want to use those instead? https://github.com/youtube/cobalt/blob/29e0d650bb4b88431aed9f4afcb983b41dd19ce3/base/metrics/histogram_macros.h#L329

Comment (at-ninja) [reply cobalt/dom/source_buffer_metrics.cc:140]: These look to be for sizes much larger than this data is expected to be.

`UMA_HISTOGRAM_MEMORY_KB` says it's for memory stats of size 500,000kb to 500M.

From local testing, I expect the sizes here to be anywhere between a few hundred bytes to ~10000kb at the maximum.

File: cobalt/dom/source_buffer_metrics.cc [hunk 2]
```diff
@@ -121,21 +121,33 @@ void SourceBufferMetrics::EndTracking(SourceBufferMetricsAction action,
...
+          "Cobalt.Media.SourceBuffer.AppendBuffer.BytesAppended",
+          size_appended);
+
+      int64_t bytes_per_microsecond =
+          size_appended / action_duration.InMicroseconds();
```
Comment (xiaomings) [inline cobalt/dom/source_buffer_metrics.cc:145]: Consider guarding `action_duration` against 0.  Also, consider moving the extra pair of {} around the calculation and log of `bytes_per_microsecond` so it's more "local".

Comment (at-ninja) [reply cobalt/dom/source_buffer_metrics.cc]: Done.

File: cobalt/dom/source_buffer_metrics.cc [hunk 3]
```diff
@@ -121,21 +121,36 @@ void SourceBufferMetrics::EndTracking(SourceBufferMetricsAction action,
...
+      if (action_duration.InMicroseconds() > 0) {
+        int64_t bytes_per_microsecond =
+            size_appended / action_duration.InMicroseconds();
+        UMA_HISTOGRAM_COUNTS_100000(
+            "Cobalt.Media.SourceBuffer.AppendBuffer."
+            "BytesAppendedPerMicrosecond",
+            bytes_per_microsecond);
+      }
       break;
+    }
```
Comment (xiaomings) [inline cobalt/dom/source_buffer_metrics.cc:153]: nit: Do we still need this extra pair of brackets?

Comment (at-ninja) [reply cobalt/dom/source_buffer_metrics.cc]: Removed.

**Review by joeltine — APPROVED** [review]

**Review by xiaomings — APPROVED** [review]

**Pull request merged**